compiler/x/i386/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3620,11 +3620,13 @@ TR::Register *OMR::X86::I386::TreeEvaluator::lternaryEvaluator(TR::Node *node, T
    TR::Register *falseReg = cg->evaluate(falseVal);
    TR::Register *trueReg  = cg->longClobberEvaluate(trueVal);
 
-   TR::ILOpCodes condOp = condition->getOpCodeValue();
-   if((condOp == TR::icmpeq) || (condOp == TR::icmpne))
+   auto condOp = condition->getOpCode();
+   bool longCompare = (condition->getOpCode().isBooleanCompare() &&
```

Comment:
again not sure about the boolean compare here

Reply:
As discussed in previous comment, we cannot assume first child is always a boolean compare.